sources/osg/StateSet.js

```diff
@@ -167,14 +176,40 @@ StateSet.prototype = MACROUTILS.objectLibraryClass( MACROUTILS.objectInherit( Ob
     getUpdateCallbackList: function () {
         return this._updateCallbackList;
     },
+
     removeUpdateCallback: function ( cb ) {
-        var arrayIdx = this._updateCallbackList.indexOf( cb );
-        if ( arrayIdx !== -1 )
-            this._updateCallbackList.splice( arrayIdx, 1 );
+        var idx = this._updateCallbackList.indexOf( cb );
+        if ( idx === -1 ) return;
+        this._updateCallbackList.splice( idx, 1 );
+
+        if ( this._updateCallbackList.length === 0 ) {
+            var parents = this._parents;
+            for ( var i = 0, l = parents.length; i < l; i++ ) {
+                var parent = parents[ i ];
+                parent.setNumChildrenRequiringUpdateTraversal( parent.getNumChildrenRequiringUpdateTraversal() - 1 );
+            }
+        }
+    },
+
+    requiresUpdateTraversal: function () {
+        return Boolean( this._updateCallbackList.length );
```

Comment:
If the function is called many times, "!!" is prefered over "Boolean" for performance reason.
https://jsfiddle.net/m40qbpnf/ (jsperf is down...)